ANMN/burst_averaged_product/burstproduct/burst_average_product.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from generate_netcdf_att import *
+from math import isnan
+from netCDF4 import Dataset, num2date, date2num
+import numpy as np
+import os
+import pandas as pd
+import re
+import shutil
+import subprocess
+import tempfile
+import time
+
+"""TODO
+- cleaning get_input_file_rel_path, handle error
+- logging
+
+- for CTD file
+http://thredds.aodn.org.au/thredds/catalog/IMOS/ANMN/NRS/NRSKAI/CTD_timeseries/catalog.html
+- remove VOLT, TIMESERIES, NOMINAL_DEPTH????
+- add all variables to existing conf file. Should work also for WQM file
+- refactore 1 or 2 WQM function as CTD will be very similar
+"""
+
+
+def get_input_file_rel_path(input_netcdf_file_path):
+    """
+    find the relative path hierarchy of an input FV01 file. The value will be
+    used in a gatt of the burst netcdf file
+    """
+    if "DATA_SERVICES_DIR" in os.environ.keys():
+        dest_path= os.path.join(os.environ["DATA_SERVICES_DIR"], 'ANMN/common/dest_path.py')
+        if os.path.exists(dest_path):
+            file_path = subprocess.check_output("%s %s" % (dest_path, input_netcdf_file_path), shell=True).strip('\n')
+            return file_path
+        else:
+            return ''
+            # sys.exit(1)
+    else:
+        return ''
+
+def get_burst_duration(netcdf_file_path):
+    """
+    returns the burst duration as float defined in the NetCDF file if exists
+    """
+    netcdf_file_obj = Dataset(netcdf_file_path, 'r', format='NETCDF4')
```

Comment:
You could save repeatedly opening and closing the file by just opening it in the main program and passing the `Dataset` object around instead of the filename.